compiler/rustc_arena/src/lib.rs

```diff
@@ -223,10 +223,8 @@ impl<T> TypedArena<T> {
                 // If the previous chunk's len is less than HUGE_PAGE
                 // bytes, then this chunk will be least double the previous
                 // chunk's size.
-                new_cap = last_chunk.storage.len();
-                if new_cap < HUGE_PAGE / elem_size {
-                    new_cap = new_cap.checked_mul(2).unwrap();
-                }
+                new_cap = last_chunk.storage.len().min(HUGE_PAGE / 2);
```

Comment:
you now use `HUGE_PAGE` instead of `HUGE_PAGE / elem_size` here, is that intentional?

Reply:
Smells like a copy-paste error, thanks!